sdk/iOS/src/MSSyncContext.m

```diff
@@ -404,54 +406,53 @@ - (void) pullWithQuery:(MSQuery *)query queryId:(NSString *)queryId completion:(
-        // Before we can pull from the remote, we need to make sure out table doesn't having pending operations
-        NSArray *tableOps = [self.operationQueue getOperationsForTable:query.table.name item:nil];
-        if (tableOps.count > 0) {
-            [self pushWithCompletion:^(NSError *error) {
-                // For now we just abort the pull if the push failed to complete successfully
-                // Long term we can be smarter and check if our table succeeded
-                if (error) {
-                    if (completion) {
-                        [self.callbackQueue addOperationWithBlock:^{
-                            completion(error);
-                        }];
-                    }
-                } else {
-                    // Check again if we have new pending ops while we synced, and repeat as needed
-                    [self pullWithQueryInternal:query queryId:queryId maxRecords:maxRecords completion:completion];
-                }
-            }];
-            return;
-        } else {
-            // TODO: Allow users to cancel operations
-            MSQueuePullOperation *pull = [[MSQueuePullOperation alloc] initWithSyncContext:self
-                                                                                     query:query
-                                                                                   queryId:queryId
-                                                                                maxRecords:maxRecords
-                                                                             dispatchQueue:writeOperationQueue
-                                                                             callbackQueue:self.callbackQueue
-                                                                                completion:completion];
-            
-            
-            [pushQueue_ addOperation:pull];
-        }
-    });
+	MSQueuePullOperation *pull = [[MSQueuePullOperation alloc] initWithSyncContext:self
+																			 query:query
+																		   queryId:queryId
+																		maxRecords:maxRecords
+																	 dispatchQueue:writeOperationQueue
+																	 callbackQueue:self.callbackQueue
+																		completion:completion];
+	
+	// Before we can pull from the remote, we need to make sure out table doesn't having pending operations
+	NSArray *tableOps = [self.operationQueue getOperationsForTable:query.table.name item:nil];
+	if (tableOps.count > 0) {
+		MSQueuePushOperation *push = [self pushWithCompletion:^(NSError *error) {
+			// For now we just abort the pull if the push failed to complete successfully
+			// Long term we can be smarter and check if our table succeeded
+			if (error) {
+				if (completion) {
+					[self.callbackQueue addOperationWithBlock:^{
+						completion(error);
+					}];
+				}
+			} else {
+				// Check again if we have new pending ops while we synced, and repeat as needed
+				[pushQueue_ addOperation:pull];
```

Comment:
I'm OK with the current logic, and only doing 1 push for now, after I wrote this, I ended up adding code to handle pending ops post pull anyways, so its not really necessary to even do the 1 push.  In addtion cancelling the pull NSOperation, should cancel the dependent Push, so this whole section may need reworked later on anyways.  For now, just fix the docs on this function to indicate 1 push will be attempted if pending ops exist for the table.